rust/main/agents/relayer/src/server/message_retry.rs

```diff
@@ -1,32 +1,89 @@
+use std::sync::Arc;
+
 use crate::settings::matching_list::MatchingList;
+
 use axum::{extract::State, routing, Json, Router};
+
 use derive_new::new;
-use tokio::sync::broadcast::Sender;
+use serde::{Deserialize, Serialize};
+use tokio::sync::{broadcast::Sender, mpsc, Mutex};
 
 const MESSAGE_RETRY_API_BASE: &str = "/message_retry";
 
-#[derive(new, Clone)]
+#[derive(new)]
 pub struct MessageRetryApi {
-    tx: Sender<MatchingList>,
+    tx: Sender<MessageRetryRequest>,
+    rx: Arc<Mutex<mpsc::Receiver<MessageRetryResponse>>>,
+}
+
+#[derive(Clone, Debug)]
+pub struct MessageRetryApiState {
+    pub tx: Sender<MessageRetryRequest>,
+    pub rx: Arc<Mutex<mpsc::Receiver<MessageRetryResponse>>>,
+}
+
+#[derive(Clone, Debug)]
+pub struct MessageRetryRequest {
+    pub uuid: String,
+    pub pattern: MatchingList,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct MessageRetryResponse {
+    /// ID of the retry request
+    pub uuid: String,
+    /// how many pending operations were processed
+    pub processed: usize,
```

Comment:
nit: would probably also avoid `processed` as we use this jargon elsewhere in lots of places, bound to get confusing

Reply:
What would you propose as a better name?

Reply:
idk maybe `evaluated` or something?